Vacant/improvements

aneves opened the pull request:
### All Submissions:

- [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Tested Changes:

1. [x] Have you lint your code (lua lint) locally prior to submission?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully ran tests with your changes locally?

### Comments

This adds a toggle to the admin UI to enable or disable the functionality, to allow for more confidence in including it on long-running games like Journey.

Linted the module.

---

This replaces PR #377 - first part, polishing the module.
The second part lands on top of this - patching Journey to use this module (but only when it is required externally).

aneves committed:
Admins can disable clear-vacant-players.

Admins get a toggle in the Config to toggle the module
clear-vacant-players off and on, if it is loaded into the map.

aneves committed:
Uniformize table.remove and table.insert in module

Use table.remove() and table.insert() uniformly in module
clear_vacant_players.

aneves committed:
Lint changes. 🧙

Fully linted the clear_vacant_players file.

Gerkiz committed:
Merge branch 'develop' into vacant/improvements

Gerkiz commented on modules/clear_vacant_players.lua:
```diff
@@ -142,36 +166,40 @@ function Public.set_offline_players_surface_removal(value)
 end
 
-local remove_offline_players = Public.remove_offline_players
+local function should_ignore_surface(surface)
```
Is this supposed to be here? It looks like an empty function that is missing the ``end`` statement

aneves replied:
No, it's an artifact from rebasing/re-sorting.
It should be only in 8a8b52ab. :/

Fixed in 695a513e.

aneves committed:
Remove empty function.

Should be in 8a8b52ab of branch vacant/add-to-journey, got wrongly included here when rebasing.

aneves committed:
Merge branch 'develop' into vacant/improvements
